util/src/main/java/io/kubernetes/client/Copy.java

```diff
@@ -164,6 +172,120 @@ public void copyDirectoryFromPod(
     }
   }
 
+  // This creates directories and files using tree of files and directories under container
+  private void createDirectoryStructureFromTree(
+      TreeNode tree,
+      String namespace,
+      String pod,
+      String container,
+      String srcPath,
+      Path destination)
+      throws IOException, ApiException {
+    // Code for creating invidual directory and files
+    createDirectory(tree, destination);
+    createFiles(tree, namespace, pod, container, srcPath, destination);
+  }
+
+  // Method to create files from directories/files tree in destination
+  private void createFiles(
+      TreeNode node,
+      String namespace,
+      String pod,
+      String container,
+      String srcPath,
+      Path destination)
+      throws IOException, ApiException {
+    if (node == null) {
+      return;
+    }
+    for (TreeNode childNode : node.getChildren()) {
+      if (!childNode.isDir) {
+        // Create file which is under 'node'
+        String filePath = genericPathBuilder(destination.toString(), childNode.name);
+        File f = new File(filePath);
+        if (!f.createNewFile()) {
+          throw new IOException("Failed to create file: " + f);
+        }
+        String modifiedSrcPath = genericPathBuilder(srcPath, childNode.name);
+        try (InputStream is = copyFileFromPod(namespace, pod, modifiedSrcPath);
+            OutputStream fs = new FileOutputStream(f)) {
+          ByteStreams.copy(is, fs);
+          fs.flush();
+        }
+      } else {
+        String newSrcPath = genericPathBuilder(srcPath, childNode.name);
+        // TODO: Change this once the method genericPathBuilder is changed to varargs
+        Path newDestination =
+            Paths.get(destination.toString() + "/" + node.name + "/" + childNode.name);
```

Comment:
`Paths.get(destination.toString(), node.name, childNode.name);` to ensure it works on Windows.

Reply:
I have changed it.